### DISCUSSION TITLE
Huawai Watch 2 not working as collector

### DISCUSSION THREAD

**Discussion by becksen** [NightscoutFoundation/xDrip#1971] [Q&A]
Hi

I've followed the tutorial https://github.com/NightscoutFoundation/xDrip/wiki/Patching-Android-Wear-devices-for-use-with-the-G5 and succeeded to install.

Couldn't find XDrip in the "apps on your phone".

I've downgraded the watch to OWDD.180215.018 upfront

Redoing the whole process I can see on the watch in the TWRP logs the following errors.
ERROR: with firmware patch A
ERROR: with firmware patch B
ERROR: could not patch BT
ERROR: with demigod mode
Demigod-2 mode enabled!

Can you advise how to successfully apply the patches and activate demigod mode?

Regards

**Comment by Navid200** [comment]
Have you seen this?
https://navid200.github.io/xDrip/docs/WearOS-Instructions.html

**Reply by becksen** [reply]
Not yet. Let me check.
Thanks for the quick reply.

**Reply by becksen** [reply]
I've tried it. As the post is quite old I've used xDrip-plus-20211004-49817a5.apk this apk and installed on phone as well as on watch.
When I go to Xdrip prefs on the watch I can see that DemiGod Mode is checked.

When I run Force update  Version: 49817a5-2021.10.04. When I start it is seems to running. It shows Xdrip+ update percentage. But after restarting the watch the "force update" entry stays unchanged. So I can to it again and again but the force update does not disappear.

I still don't get any connection to my Dexcom G6 transmitter. 

The Logs are saying:
No valid timestamp stored for transmitter: <transmitter id>

Any suggestions?
Thanks.

Update:
Log:
Wearob1G5 StateMachine:
Could not read after AutRequetTX: complidea.rxandroidble2.exceptions.BleGattCannotStartException
Authentication notification throwable: (CHECK_AUTH)
(GATT_CONN_TERMINAT_PEER_USER)

**Reply by becksen** [reply]
Updated to latest nightly build: f43d037-2202.01.28

Log:
`waerjamorham excec
Task deadlock on: sendData! @4
Task deadlock on: sendData! @5
Task deadlock on: sendData! @6
Task deadlock on: sendData! @7
Task deadlock on: sendData! @8
Task deadlock on: sendData! @9
Task deadlock on: sendData! @10
Task deadlock on: sendData! @11
Task deadlock on: sendData! @8
Task deadlock on: sendData! @9
Task deadlock on: sendData! @7`

In Watch bluetooth menu I can see DexcomPP device which is not available. It seems that what might have had short connection to the Dexcom G6 Transmitter.

Log:
`wearOb1G5StateMachine:
Stor: VersionRX dbg: 53000A0F000000000000303235302D50726F771A
Wrote version Request: 2
Coud not read after AuthRequestTX: com.polidea.rxandroidble2.exeptions.BleGattCannotStartExceoption: GATT exceoption from MAC address D1:15:62:8A:D0:32, with type BletGattOpetion(Description='CHARACTERISTIC_READ#9
Could not read reply to auth challenge:....
Cannot process backfill evaluation as no active sensor`

**Reply by becksen** [reply]
I'm using a 8L transmitter

**Comment by becksen** [comment]
Build: OWDD.180215.018
Google WearOS 2.0
Android 8.0.0
